src/lib/non_zero_curve_point/non_zero_curve_point.ml

```diff
@@ -2,71 +2,50 @@
 "/src/config.mlh"]
 
 open Core_kernel
-open Snark_params
 open Module_version
+module T = Functor.Make (Snark_params.Tick)
 
-let parity y = Tick.Bigint.(test_bit (of_field y) 0)
-
-let gen_uncompressed =
-  Quickcheck.Generator.filter_map Tick.Field.gen_uniform ~f:(fun x ->
-      let open Option.Let_syntax in
-      let%map y = Tick.Inner_curve.find_y x in
-      (x, y) )
+(* the Make functor does not supply any bin_prot functions, so 
+   we define them here. The base58 and yojson definitions 
+   depend on those functions, so they're also in this file
+ *)
 
 module Compressed = struct
-  open Tick
+  open Compressed_poly
 
-  module Poly = struct
+  module Arg = struct
+    (* module with same type t as Stable below, to give as functor argument *)
```

Comment:
I think this is okay for now, but we should update the module versioning ppx to support this better.